### ISSUE TITLE
Disconnect without connection status callback called

### ISSUE THREAD

**Issue by Patrik-Berglund** [Azure/azure-iot-sdk-c#1255]
Development Machine, OS, Compiler (and Other Relevant Toolchain Info)
ESP-IDF v3.3
Compiled on Ubuntu 16.04 using GCC 5.4.0

SDK Version (Please Give Commit SHA if Manually Compiling)
Release 2019-07-01

Protocol
MQTT
Using tlsio_mbedtls and socketio_berkeley from c-utility

Describe the Bug
Connection status callback not called on disconnect after "IOTHUB_CLIENT_CONFIRMATION_MESSAGE_TIMEOUT". The client just reconnects without notification.

Error log,
```
I (133463) IOT HUB: Result Call Back Called! Result is: IOTHUB_CLIENT_CONFIRMATION_MESSAGE_TIMEOUT 
Error: Time:Fri Oct 11 09:15:38 2019 File:/components/azure-iot/pal/src/socketio_berkeley.c Func:socketio_send Line:834 Failure: socket state is not opened.
-> 09:15:38 DISCONNECT
Error: Time:Fri Oct 11 09:15:38 2019 File:/components/azure-iot/pal/src/socketio_berkeley.c Func:socketio_send Line:834 Failure: socket state is not opened.
-> 09:15:39 CONNECT | VER: 4 | KEEPALIVE: 60 | FLAGS: 192 | USERNAME: XXXXXXX.azure-devices.net/XXXXXXX/?api-version=2017-11-08-preview&DeviceClientType=iothubclient%2f1.3.4%20(berkeley_mbedtls) | PWD: XXXX | CLEAN: 0
<- 09:15:40 CONNACK | SESSION_PRESENT: true | RETURN_CODE: 0x0
I (135153) IOT HUB: Connection Status Callback: status=0 reason=6
I (135153) IOT HUB: IOTHUB_CLIENT_CONNECTION_OK
```

**Comment by ewertons** [comment]
Thank you, @Patrik-Berglund !

This indeed seems incorrect, we will verify and prioritize it accordingly.

**Comment by momuno** [comment]
Hi @Patrik-Berglund, I am looking into this bug.  Can you please tell me what device you were using? Was it an esp2866, esp32, or something else?

**Comment by Patrik-Berglund** [comment]
Hi @momuno 

I don't work for the company anymore where this product was developed.

At the time being the esp-azure library wasn't that mature, so it was better to use what you have build instead using mbedtls and socket io berkeley. By doing that you get support for all protocols (http, mqtt and amqp + websockets) and all features like blob upload, DPS and proven retry logic and so on...

The error was probably not related to the ESP-IDF, so I think you should be able to reproduce this using ubuntu and the LL api.

The problem is I don't remember how to trigger a "IOTHUB_CLIENT_CONFIRMATION_MESSAGE_TIMEOUT" fault. Since it was more than a year ago I posted this...

**Comment by momuno** [comment]
Hi @Patrik-Berglund no worries!  Thank you so much for taking the time to respond.  I appreciate your feedback.

**Comment by momuno** [comment]
Source of bug and fix described in PR #1735

**Comment by az-iot-builder-01** [comment]
@Patrik-Berglund, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
